### ISSUE TITLE
Create wheelbarrow for loading and hauling rocks, regoliths, ores, minerals, etc. 

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#713]
**Is your feature request related to a problem? Please describe.**
- It makes more sense to create a container called wheelbarrow for loading and transporting excavated resources 

**Describe the solution you desire to see**
- Each wheelbarrow does not need much distinguishing characteristics from others
- Each wheelbarrow does need to track the amount of resources content loaded inside and who the owner is
- Create a new class that extends `Part` or `ItemResource`
- Create a list inside this new class for tracking a list of wheelbarrows

**Describe the rationale**
- A large bag or a bag was previously used in Mining, CollectMinedResource and DigLocal as the container fore excavated resources. 
- However a person has to carry the bag and there's a weight limitation imposed upon the person.
- The use of wheelbarrow would be an appropriate replacement for bags
- However, based on the discussion of [bottle](https://github.com/mars-sim/mars-sim/issues/712), we should experiment with creating a new container class that doesn't extends `Equipment` or `Unit`.

**Describe alternatives you've considered**
- Based on our discussion at this [thread](https://github.com/mars-sim/mars-sim/commit/cc0ec76c14d18146dc1c2c97b06a0745ef93fd0c#r85634370), wheelbarrow could be an equipment.
- However, treating each wheelbarrow as a GenericContainer means that it would extend Equipment and each would be an individual objects.

**Specifications  (please complete the following information):**
 - pre-3.4.0 build 7457

**Comment by bevans2000** [comment]
I think that wheelbarrow should follow the current Container approach for now. Then we convert all of them into a new model in one operation.

**Comment by mokun** [comment]
I've tried creating a new `StorableItem` class that extends Part. 

But to make it truly work, I do need to have a separate manager class to distinguish one wheelbarrow from another. In that case, it does seem to involve more changes.

So if it's not workable, I'll go back to making wheelbarrow an EquipmentType.

**Comment by bevans2000** [comment]
Yes you could end up just reproducing the Unitmanager class. We have a clear set of interfaces of ResourceHolder and EquipmentOwner so we should work within that framework then convert Equipment. 
Sounds like the need for a project to sketch it out.

**Comment by mokun** [comment]
Now that we have put wheelbarrow into use, how do you make a person *push* a wheelbarrow, instead of *carrying* it ?  

A wheelbarrow itself weighs 12 kg. For now, define the total capacity is defined as 100 kg. 

The advantage of a wheelbarrow is supposed to make things easy for settlers and save time and effort.

Even when fully loaded, since it has wheels, a person with low strength and endurance should be able to push it along. 

What would you change in order to lower the requirement and simulate this ***towing*** effect ?

**Comment by bevans2000** [comment]
So it's all about the carrying mass calculation. I think simulating the towing or pushing is not needed; we just assume if a Wheelbarrow is assigned to a Person then it's used. How is is moved is too low level an action to simulate.

I see it as

- When assigned to a Vehicle or Settlement - the contributing mass follows the normal existing mass logic, full mass
- When assigned to a Person/Robot - the contributing mass would be a significantly reduced percentage since they only have a small amount to lift the wheelbarrow

However care is needed in the implementation to avoid the Wheel Barrow concept everywhere which would be wrong. One way is to change the Equipment.getMass method. If the container is a Person then apply a modifier to the mass returned. The modifier would be chosen according to the EquipmentType where wheelbarrow has a low value. This is easy but not a good model.

Probably the perfect way is to have a new method getContributingMass on Equipment that has the same logic as above where a modified is used. This new method is used when calculating the total carrying mass of an EquipmentInventory. This method is better because it preserves the mass of the actual Equipment. EquipmentInventory.getStoredmass is the key location where a different mass must be returned.

To avoid saving the modifier on every Equipment instance I would select it from the Containerutil class.

**Comment by mokun** [comment]
> to have a new method getContributingMass on Equipment

Yes. We only need to consider the **effect** of adding this *modified* mass onto a person (as if a person is physically carrying a percentage of the wheelbarrow and together with whatever is loaded onto it).

Currently, this will affect the walking speed of a person and nothing else.

By the way, the walking speed calculation was buggy previously. Hope it's been fixed.

**Comment by mokun** [comment]
I'm trying to come up with getModifiedMass() and/or modify calculateWalkSpeed() in Person as we speak.

This is just to speed up the walking speed by introducing the adjusted mass to simulate the idea of *pushing* a wheelbarrow, instead of carrying a wheelbarrow, especially when it's loaded.

```
	/**
	 * Calculates the modifier for walking speed based on how much this unit is carrying.
	 */
	public double calculateWalkSpeed() {
		double mass = getMass();
		// At full capacity, may still move at 10%.
		// Make sure is doesn't go -ve and there is always some movement
		return 1.1 - Math.min(mass/Math.max(carryingCapacity, SMALL_AMOUNT), 1D);
	}
```

**Comment by bevans2000** [comment]
Why not gave a ```getCarryingMass``` method on Person and use that just for the walking calculate?
